app/gui/src/project-view/stores/suggestionDatabase/index.ts

```diff
@@ -80,6 +80,13 @@ export class SuggestionDb extends ReactiveDb<SuggestionId, SuggestionEntry> {
     )
   }
 
+  /** Returns methods defined on the specified type, including private methods. */
+  typeMethods(memberOf: ProjectPath): IterableIterator<MethodSuggestionEntry> {
+    return iter.filter(this.getAllEntriesOfKind(SuggestionKind.Method), (method) =>
+      memberOf.equals(method.memberOf),
+    )
```

Comment:
What about making a new ReactiveIndex?

Reply:
I'm not sure why?

The approach here should perform better. A `ReactiveIndex` leaks memory (accumulates effects without flushing) when it receives ongoing updates (SDB changes), but may not be used (only when a table expression is edited). Without an index the effects belong to a `computed`, which is eagerly dirtied so its memory requirements are bounded.

If we had any SDB data consumer that would make use of a methods index that would react to it continuously (or at least frequently), then `ReactiveIndex` may share some work (and wouldn't leak memory); but we have so far not needed that specific information anywhere.